arm_gdb/mpu.py

```diff
@@ -0,0 +1,248 @@
+                (0b0111, None, "Normal Memory, Inner Write-Back Transient, Allocate RW", None),
+                (0b1000, None, "Normal Memory, Inner Write-Through Non-transient", None),
+                (0b1001, None, "Normal Memory, Inner Write-Through Non-transient, Allocate W", None),
+                (0b1010, None, "Normal Memory, Inner Write-Through Non-transient, Allocate R", None),
+                (0b1011, None, "Normal Memory, Inner Write-Through Non-transient, Allocate RW", None),
+                (0b1100, None, "Normal Memory, Inner Write-Back Non-transient", None),
+                (0b1101, None, "Normal Memory, Inner Write-Back Non-transient, Allocate W", None),
+                (0b1110, None, "Normal Memory, Inner Write-Back Non-transient, Allocate R", None),
+                (0b1111, None, "Normal Memory, Inner Write-Back Non-transient, Allocate RW", None),
+            ], "Device attributes. Specifies the memory attributes for Device. (Valid only if Outer == 0)")
+        ]
+    return [
+        RegisterDef("MPU_TYPE", "MPU Type Register", 0xE000ED90, 4, [
+            FieldBitfield("DREGION",  8, 8,
+                        "Number of regions supported by the MPU."),
+            FieldBitfield("SEPARATE",  0, 1,
+                        "Indicates support for separate instructions and data address regions"),
+        ]),
+        RegisterDef("MPU_CTRL", "MPU Control Register", 0xE000ED94, 4, [
+            FieldBitfield("PRIVDEFENA", 2, 1,
+                        "Privileged default enable. Controls whether the system address map is enabled for privileged software"),
+            FieldBitfield("HFNMIENA", 1, 1,
+                        "HardFault, NMI enable. Controls whether handlers executing with a requested execution priority of less than 0 access memory with the MPU enabled or disabled"),
+            FieldBitfield("ENABLE", 0, 1,
+                        "Enable. Enables the MPU."),
+        ]),
+        RegisterDef("MPU_RNR", "MPU Region Number Register", 0xE000ED98, 4, [
+            FieldBitfield("REGION", 0, 8,
+                        "Region number. Indicates the memory region accessed by MPU_RBAR and MPU_RLAR."),
+        ]),
+        RegisterDef("MPU_MAIR0", "MPU Memory Attribute Indirection Register 0", 0xE000EDC0, 4,
+                    mair_attr_fields(0, 0) +
+                    mair_attr_fields(1, 8) + 
+                    mair_attr_fields(2, 16) + 
+                    mair_attr_fields(3, 24)
+        ),
+        RegisterDef("MPU_MAIR1", "MPU Memory Attribute Indirection Register 0", 0xE000EDC4, 4,
+                    mair_attr_fields(4, 0) +
+                    mair_attr_fields(6, 16) + 
+                    mair_attr_fields(5, 8) + 
+                    mair_attr_fields(7, 24)
+        )
+    ]
+
+def get_mpu_region_regs(model):
+    return [
+        RegisterDef(f"MPU_RBAR", f"MPU Region Base Address Register", 0xE000ED9C, 4, [
+            FieldBitfield("BASE", 5, 27,
```

Comment:
Now the base shows like:

MPU_RBAR                         = 00500006 
    BASE                           0050000. - 0028000

in fact the address is still 0x50000,  shows 0x28000 maybe confused.

Reply:
Oh right, good catch. Got to set that as a Map field then